dask/bag/core.py

```diff
@@ -412,12 +410,11 @@ def frequencies(self, split_every=None):
         >>> dict(b.frequencies())  # doctest: +SKIP
         {'Alice': 2, 'Bob', 1}
         """
-        return self.reduction(compose(list, dictitems, frequencies),
-                              merge_frequencies,
+        return self.reduction(frequencies, merge_frequencies,
                               out_type=Bag, split_every=split_every,
-                              name='frequencies')
+                              name='frequencies').map_partitions(dictitems)
```

Comment:
The speedup comes from returning a dict for the intermediates. Only the final reduction in the tree is converted to a list, to maintain consistency.